src/pages/ProductPage.js

```diff
@@ -21,30 +22,36 @@ const ProductPage = () => {
   let itemId = useParams().id;
   const [data, setData] = useState({ products: [] });
   const [isFetching, setIsFetching] = useState(false);
+
+  const [message, setMessage] = useState('');
   useEffect(() => {
     const fetchData = async () => {
       try {
         setIsFetching(true);
-        setData({ products: data.products, isFetching: true });
+        setMessage('Loading...');
+        setData({ products: data.products });
         const response = await axios.get(API_BASE_URL + itemId);
         setIsFetching(false);
-
-        setData({ products: [response.data], isFetching: false });
+        setData({ products: [response.data] });
```

Comment:
You don't need to store this as an array, since there is only one object coming back from the request.

Reply:
fixed